File: experiment/websites/js/towersofhanoi/towersofhanoi.js
/* based on: https://rosettacode.org/wiki/Towers_of_Hanoi#ES5 (2021-12-30) */

/**
 * @param {number} n
 * @param {number|string} a
 * @param {number|string} b
 * @param {number|string} c
 */
function move(n, a, b, c) {
  if (n > 0) {
    move(n-1, a, c, b);
    //console.log("Move disk from " + a + " to " + c);
    move(n-1, b, a, c);
  }
}

File: experiment/websites/c/towersofhanoi/towersofhanoi.c
/* based on: https://rosettacode.org/wiki/Towers_of_Hanoi#C (2021-12-30) */

#include <emscripten/emscripten.h>

EMSCRIPTEN_KEEPALIVE
void move(int n, int from, int via, int to)
{
  if (n > 1) {
    move(n - 1, from, to, via);
    move(n - 1, via, from, to);
  }
}
